lib/resty/rediscluster.lua

```diff
@@ -29,21 +29,34 @@ local DEFAULT_CONNECTION_TIMEOUT = 1000
 local DEFAULT_SEND_TIMEOUT = 1000
 local DEFAULT_READ_TIMEOUT = 1000
 
+local LEFT_BRACKET = "{"
+local RIGHT_BRACKET = "}"
+
+local _M = {}
+
+local mt = { __index = _M }
+
+
+-- According to the Redis documentation, the hash tag is found only if the '{' and '}'
+-- are exist in the key, or it should use the whole key to calculate the slot.
+-- See: https://redis.io/docs/latest/operate/oss_and_stack/reference/cluster-spec/#hash-tags
+--
+-- For the Redis hash slot implemetation, see: https://github.com/redis/redis/blob/8a05f0092b0e291498b8fdb8dd93355467ceab25/src/cluster.c#L30-L49
 local function parse_key(key_str)
-    local left_tag_single_index = string_find(key_str, "{", 0)
-    local right_tag_single_index = string_find(key_str, "}", 0)
-    if left_tag_single_index and right_tag_single_index then
-        --parse hashtag
-        return key_str.sub(key_str, left_tag_single_index + 1, right_tag_single_index - 1)
-    else
+    local left_index = string_find(key_str, LEFT_BRACKET, 1, true)
+    if not left_index then
         return key_str
     end
-end
-
 
-local _M = {}
+    local right_index = string_find(key_str, RIGHT_BRACKET, left_index + 1, true)
+    if right_index and right_index > left_index + 1 then
```

Comment:
here I think we could start looking for `}` at `left_index + 2` and avoid then the `right_index > left_index + 1` check

Reply:
@samugi I intend to search from left_index+1. The reason is to align with Redis codebase implementation.